dart/lib/src/protocol/dsn.dart

```diff
@@ -14,34 +14,36 @@ class Dsn {
   final String publicKey;
 
   /// The Sentry.io secret key for the project.
-  final String secretKey;
+  final String? secretKey;
 
   /// The ID issued by Sentry.io to your project.
   ///
   /// Attached to the event payload.
   final String projectId;
 
   /// The DSN URI.
-  final Uri uri;
+  final Uri? uri;
 
-  String get postUri {
-    final port = uri.hasPort &&
-            ((uri.scheme == 'http' && uri.port != 80) ||
-                (uri.scheme == 'https' && uri.port != 443))
-        ? ':${uri.port}'
+  Uri get postUri {
+    final port = uri!.hasPort &&
```

Comment:
can we create a new local field like `final uriCopy = uri!` and using only `uriCopy`?
calling `uri!` every time affects readability